lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -243,7 +242,7 @@ def create_post_for_auth_or_purchase(money, payment, options)
           add_customer_data(post, options)
           add_metadata(post, options)
           post[:description] = options[:description]
-          post[:statement_description] = options[:statement_description]
+          post[:statement_descriptor] = options[:statement_description]
```

Comment:
I'm a little nervous about this change, but the other changes in here look ok - doing a little more testing to confirm.